flink-streaming-java/src/main/java/org/apache/flink/streaming/api/datastream/DataStreamSource.java

```diff
@@ -44,6 +44,11 @@ public DataStreamSource(StreamExecutionEnvironment environment,
 		}
 	}
 
+	public DataStreamSource(SingleOutputStreamOperator<T> operator) {
```

Comment:
Here, we should always set `isParallel` to `true`. It is not quite obvious but the field is used to disallow changing the parallelism for a `SourceFunction` that cannot be parallelized. Our new operator can always run in parallel.